plugins/modules/nd_remote_storage_location.py

```diff
@@ -0,0 +1,333 @@
+
+    if state == "present":
+        if not nd.existing:
+            payload = {
+                "name": name,
+                "description": description,
+                "port": server_port,
+                "hostname": server_name,
+                "path": remote_path,
+            }
+
+            if nas and not check_if_all_elements_are_none(nas):
+                payload["type"] = "nfs"
+            elif sftp_scp and sftp_scp.get("protocol"):
+                payload["type"] = sftp_scp.get("protocol")
+
+            # NAS create
+            if payload.get("type") == "nfs":
+                payload["alertThreshold"] = nas.get("alert_threshold")
+                payload["limit"] = nas.get("limit")
+                payload["readWrite"] = nas.get("read_write")
+            # SFTP/SCP create
+            elif payload.get("type") == "sftp" or payload.get("type") == "scp":
+                payload["authentication"] = dict()
+                if sftp_scp.get("password"):
+                    payload["authentication"]["type"] = "password"
+                    payload["authentication"]["username"] = sftp_scp.get("username")
+                    payload["authentication"]["password"] = sftp_scp.get("password")
+                elif sftp_scp.get("ssh_key"):
+                    payload["authentication"]["type"] = "key"
+                    payload["authentication"]["username"] = sftp_scp.get("username")
+                    payload["authentication"]["sshKey"] = sftp_scp.get("ssh_key")
+                    payload["authentication"]["passphrase"] = sftp_scp.get("passphrase")
+
+        else:
+            payload = copy.deepcopy(nd.existing)
+            payload["description"] = description or payload.get("description")
+
+            # SFTP/SCP update
+            if sftp_scp:
+                payload["type"] = sftp_scp.get("protocol") or payload.get("type")
+                payload["path"] = remote_path or payload.get("path")
+                if sftp_scp.get("password"):
+                    payload["authentication"]["type"] = "password"
+                    payload["authentication"]["password"] = sftp_scp.get("password")
+                elif sftp_scp.get("ssh_key"):
+                    payload["authentication"]["type"] = "key"
+                    payload["authentication"]["sshKey"] = sftp_scp.get("ssh_key")
+                    payload["authentication"]["passphrase"] = sftp_scp.get("passphrase")
+
+            # NAS update
+            elif payload.get("type") == "nfs":
+                payload["limit"] = nas.get("limit") or payload.get("limit")
+                payload["readWrite"] = nas.get("read_write") if nas.get("read_write") is not None else payload.get("readWrite")
+
+        delete_none_values(payload, recursive=True)
+        nd.sanitize(payload, collate=True)
+
+        if not module.check_mode:
+            if nd.existing:
+                nd.request(path, method="PUT", data=payload)
```

Comment:
Is PUT op called even for the same configurations?
If yes, shall we leverage get_diff()?